x-pack/plugins/enterprise_search/public/applications/workplace_search/views/role_mappings/user.test.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import '../../../__mocks__/react_router';
+import '../../../__mocks__/shallow_useeffect.mock';
+import { setMockActions, setMockValues } from '../../../__mocks__/kea_logic';
+import { groups } from '../../__mocks__/groups.mock';
+
+import React from 'react';
+
+import { shallow } from 'enzyme';
+
+import { UserFlyout, UserAddedInfo, UserInvitationCallout } from '../../../shared/role_mapping';
+import { elasticsearchUsers } from '../../../shared/role_mapping/__mocks__/elasticsearch_users';
+import { wsSingleUserRoleMapping } from '../../../shared/role_mapping/__mocks__/roles';
+
+import { GroupAssignmentSelector } from './group_assignment_selector';
+import { User } from './user';
+
+describe('User', () => {
+  const handleSaveUser = jest.fn();
+  const closeUsersAndRolesFlyout = jest.fn();
+  const setUserExistingRadioValue = jest.fn();
+  const setElasticsearchUsernameValue = jest.fn();
+  const setElasticsearchEmailValue = jest.fn();
+  const handleRoleChange = jest.fn();
+  const handleUsernameSelectChange = jest.fn();
+
+  const mockValues = {
+    availableGroups: [],
+    singleUserRoleMapping: null,
+    userFormUserIsExisting: false,
+    elasticsearchUsers: [],
+    elasticsearchUser: {},
+    roleType: 'admin',
+    roleMappingErrors: [],
+    userCreated: false,
+    userFormIsNewUser: false,
+  };
+
+  beforeEach(() => {
+    setMockActions({
+      handleSaveUser,
```

Comment:
[not a change request, just a comment] in general in AS we put our actions in a single obj, similar to mockValues, so they don't need to be redeclared individually here:

```js
const mockActions = { a: jest.fn(), b: jest.fn() } // etc
setMockActions(mockActions);
```

just a little less repetitive is all